### PR TITLE
Adds Discord Rich Presence functionality to SaladBind

### PR DESCRIPTION
**Pull request by memedmusic**
--------- This is just copy-pasted from my failed attempt ------------
This is my first contribution on GitHub, so hopefully I've done everything right.

With this new code users can opt-in to have a discord rich presence display one of 3 different status':

In main menu
Configuring miners
Mining with {miner name} (will also display how long the user has been mining for)
Why I think this is a good idea:

Extra promotion for SaladBind
Can be used to inform a user if their miner is still running
If the user doesn't want to have a rich presence, they can turn it off at anytime using the Configure SaladBind interface

I have tested the compiled app on windows and to my knowledge it is all working fine.

### CODE REVIEW

**Review by WalkxCode — APPROVED** [review]
Walky approves 🦶

File: setup.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ async function continueSetup(clear) {
     const discordPresencePrompt = await inquirer.prompt([{
         type: 'confirm',
         name: 'presence',
-        message: chalk.yellow("Do you want to opt-in to the Saladbind Discord rich presence? (This may need an app restart)"),
+        message: chalk.yellow("Allow Discord rich presence to be shown? (May require app restart)"),
```
Comment (WalkxCode) [inline setup.js:40]: Maybe "Enable Discord Rich Presence?" would sound better?

Comment (memedmusic) [reply setup.js]: Yeah, it does. I just changed it

**Review by WalkxCode — APPROVED** [review]

**Review by vtheskeleton — CHANGES_REQUESTED** [review]
Please do not change whitespace from tabs to spaces.

File: presence.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+const RPC = require('discord-rpc');
+const client = new RPC.Client({ transport: 'ipc' });
+const fs = require("fs");
+var { config } = require('process');
```
Comment (vtheskeleton) [inline presence.js:4]: > var { config } = require('process');
What is this for?

Comment (memedmusic) [reply presence.js:4]: I have no idea. I must have left it there while I was testing stuff. I'll remove it

**Review by vtheskeleton — CHANGES_REQUESTED** [review]
Great, now just the whitespace and I can merge this

**Comment by memedmusic** [comment]
oh god what did i do aaaaaaaa

**Comment by memedmusic** [comment]
what lmao am i just stupid or is it still spaces

File: mining.js [hunk 3]
```diff
@@ -16,315 +16,314 @@ let rawdata = fs.readFileSync(tempest);
...
-	let folderData = fs.readdirSync(`./data/miners/${folderName}`)
-	if(folderData.length == 1) {
-		mv(`./data/miners/${folderName}/${folderData[0]}`, `./data/miners/${folderName}`, {clobber: false}, function(err) { //oh okers
-			// done. it tried fs.rename first, and then falls ba	ck to
-			// piping the source file to the dest file and then unlinking
-			// the source file. (docs lol) lol
-			if(err) {
-				console.log(chalk.bold.red(err));
-				spinner.fail();
-			}
```
Comment (vtheskeleton) [inline mining.js:32]: mining.js is still on spaces

**Review by vtheskeleton — CHANGES_REQUESTED** [review]
one more :(

**Comment by vtheskeleton** [comment]
i failed wtf
time to try again

**Review by vtheskeleton — APPROVED** [review]
good enough

**Comment by memedmusic** [comment]
Sorry for the extra hassle. Thanks for your help.

**Pull request merged**